LibVLCSharp/Platforms/UWP/VideoView.cs

```diff
@@ -0,0 +1,233 @@
+                $"--winrt-swapchain=0x{_swapChain.NativePointer.ToString("x")}"
+            };
+        }
+
+        void OnLoad()
+        {
+            SharpDX.DXGI.Factory2 dxgiFactory;
+            DeviceCreationFlags deviceCreationFlags = DeviceCreationFlags.BgraSupport | DeviceCreationFlags.VideoSupport;
+
+#if DEBUG
+            deviceCreationFlags |= DeviceCreationFlags.Debug;
+            try
+            {
+                dxgiFactory = new SharpDX.DXGI.Factory2(true);
+            }
+            catch (SharpDXException e)
+            {
+                dxgiFactory = new SharpDX.DXGI.Factory2(false);
+            }
+#else
+            dxgiFactory = new SharpDX.DXGI.Factory2(false);
+#endif
+            _d3d11Device = null;
+            foreach (var adapter in dxgiFactory.Adapters)
+            {
+                try
+                {
+                    _d3d11Device = new SharpDX.Direct3D11.Device(adapter, deviceCreationFlags);
+                    break;
+                }
+                catch (SharpDXException e)
+                {
+                }
+            }
+
+            if (_d3d11Device is null)
+            {
+                throw new Exception("Could not D3D11CreateDevice");
+            }
+
+            _device = _d3d11Device.QueryInterface<SharpDX.DXGI.Device1>();
+
+            //Create the swapchain
+            var swapChainDescription = new SharpDX.DXGI.SwapChainDescription1
+            {
+                Width = (int)(this._panel.ActualWidth * this._panel.CompositionScaleX),
+                Height = (int)(this._panel.ActualHeight * this._panel.CompositionScaleY),
+                Format = SharpDX.DXGI.Format.B8G8R8A8_UNorm,
+                Stereo = false,
+                SampleDescription =
+                {
+                    Count = 1,
+                    Quality = 0
+                },
+                Usage = Usage.RenderTargetOutput,
+                BufferCount = 2,
+                SwapEffect = SwapEffect.FlipSequential,
+                Flags = SwapChainFlags.None,
+                AlphaMode = AlphaMode.Unspecified
+            };
+
+            _swapChain = new SharpDX.DXGI.SwapChain1(dxgiFactory, _d3d11Device, ref swapChainDescription);
+
+            _device.MaximumFrameLatency = 1;
+
+            //TODO: perform the next 2 calls on the UI thread
+            var panelNative = ComObject.As<ISwapChainPanelNative>(this._panel);
+            panelNative.SwapChain = _swapChain;
+
+            // This is necessary so we can call Trim() on suspend
+            this._device3 = _device.QueryInterface<SharpDX.DXGI.Device3>();
+            if (_device3 == null)
+                throw new Exception();
+            this._swapchain2 = _swapChain.QueryInterface<SharpDX.DXGI.SwapChain2>();
+            if (_swapchain2 == null)
+                throw new Exception();
+
+            UpdateScale();
+            UpdateSize();
+            this._loaded = true;
+        }
+
+        void OnUnload()
```

Comment:
we need to test consecutive unloads with the same LibVLC.cs

Reply:
The method is now called `DestroySwapChain`, but is still private : The VideoView manages completely when this is called. Two consecutive unloads isn't a problem since we always check for nulls before Disposing.

Should we call `DestroySwapChain` at the beginning of the `CreateSwapChain` method? I don't think so for the same reason : it's a private member called on the `Loaded` event